static/app/components/autoComplete.tsx

```diff
@@ -132,18 +149,12 @@ class AutoComplete<T extends Item> extends Component<Props<T>, State<T>> {
     };
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps, nextState) {
+  componentDidUpdate(_prevProps: Props<T>, prevState: State<T>) {
     // If we do NOT want to close on select, then we should not reset highlight state
     // when we select an item (when we select an item, `this.state.selectedItem` changes)
-    if (!nextProps.closeOnSelect && this.state.selectedItem !== nextState.selectedItem) {
-      return;
+    if (!this.props.closeOnSelect && this.state.selectedItem !== prevState.selectedItem) {
+      this.resetHighlightState();
     }
-
-    this.resetHighlightState();
-  }
-
-  UNSAFE_componentWillUpdate() {
-    this.items.clear();
```

Comment:
This was done because we assumed every re-render would call the `getItemProps` for _every_ row, registering them in the `items`mapping. That's no longer true so we can get rid of this. They will unregister themselves.